tests/examples/examples.test.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { execSync } from "child_process";
+import { glob } from "glob";
+import { Semaphore } from "async-mutex";
+import * as fs from "fs";
+
+const pattern = process.env.EXAMPLE_NAME || `./examples/**/*.ts`;
+
+describe("E2E Examples", async () => {
+  const semaphore = new Semaphore(5);
+  const exampleFiles = await glob(pattern, {
+    cwd: process.cwd(),
+    dot: false,
+    realpath: true,
+  });
+  const config = JSON.parse(fs.readFileSync("./tests/examples/example_test.cfg", "utf8"));
+
+  for (const example of exampleFiles) {
+    if (config.exclude.includes(example)) {
+      continue;
+    }
+    it.concurrent(`Run ${example}`, async () => {
+      await semaphore.acquire();
```

Comment:
Why can't you just use https://vitest.dev/config/#sequence-concurrent with https://vitest.dev/config/#maxconcurrency? It seems better.